### PR TITLE
doc/mgr/orchestrator: Add "Service Specification"

### PR DESCRIPTION
**Pull request by sebastian-philipp**
Also:

1. doc/cephadm/drivegroups: Update to new yaml
1. doc/cephadm/drivegroup: Don't introduce DriveGroups
    
    We already have a name for it: a Service Specification of
    type `osd`. We don't need to introduce a new name for it.
    
    Well, they are "DriveGroups", but users don't need to know it.

    Note: **this is RFC and I'm happy to omit this commit**
  

Fixes: https://tracker.ceph.com/issues/44716?issue_count=93&issue_position=5&next_issue_id=44739&prev_issue_id=44598

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by sebastian-philipp** [comment]
jenkins test docs

**Comment by sebastian-philipp** [comment]
jenkins test docs

**Comment by sebastian-philipp** [comment]
jenkins render docs

File: doc/cephadm/drivegroups.rst [hunk 1]
```diff
@@ -312,21 +327,26 @@ You can use the 'host_pattern' key in the layout to target certain nodes. Salt t
...
-        model: MC-55-44-XZ
-      db_devices:
-        model: SSD-123-foo
+    data_devices:
+      model: MC-55-44-XZ
+    db_devices:
+      model: SSD-123-foo
 
-This will apply different drive groups to different hosts depending on the `host_pattern` key.
+This will apply different osd specs to different hosts depending on the `host_pattern` key.
```
Comment (asettle) [inline doc/cephadm/drivegroups.rst]:
```suggestion
This applies different OSD specs to different hosts depending on the `host_pattern` key.

File: doc/cephadm/drivegroups.rst [hunk 2]
```diff
@@ -352,18 +372,20 @@ It's however possible to deploy the WAL on a dedicated device as well, if it mak
     Size: 256GB
 
 
-The drivegroup for this case would look like this (using the `model` filter)
+The osd spec for this case would look like this (using the `model` filter)
```
Comment (asettle) [inline doc/cephadm/drivegroups.rst]:
```suggestion
The OSD spec for this case would look like the following (using the `model` filter):

File: doc/mgr/orchestrator.rst [hunk 3]
```diff
@@ -276,6 +276,76 @@ Start/stop/reload::
 
     ceph orch daemon {start,stop,reload} <type> <daemon-id>
     
+.. _orchestrator-cli-service-spec:
+    
+Service Specification
+=====================
+
+As *Service Specification* is a data structure often represented as YAML 
+to specify the deployment of services. An example might look like so::
```
Comment (asettle) [inline doc/mgr/orchestrator.rst]:
```suggestion
to specify the deployment of services. For example::

File: doc/mgr/orchestrator.rst [hunk 4]
```diff
@@ -276,6 +276,76 @@ Start/stop/reload::
...
+        - host1
+        - host2
+        - host3
+    spec: ...
+        
+Where the properties of a service specification are the following:
+
+* ``service_type`` is the type of the service. Needs to be either a Ceph
+   service (``mon``, ``crash``, ``mds``, ``mgr``, ``osd`` or 
+   ``rbd-mirror``) or a gateway (``nfs`` or ``rgw``) or part of the
```
Comment (asettle) [inline doc/mgr/orchestrator.rst]:
```suggestion
   ``rbd-mirror``), a gateway (``nfs`` or ``rgw``), or part of the

File: doc/mgr/orchestrator.rst [hunk 5]
```diff
@@ -276,6 +276,76 @@ Start/stop/reload::
...
+   monitoring stack (``alertmanager``, ``grafana``, ``node-exporter`` or
+   ``prometheus``).
+* ``service_id`` is the name of the service. Omit the service time
+* ``placement`` is a :ref:`orchestrator-cli-placement-spec`
+* ``spec``: additional specifications for a specific service.
+
+Each service type might have slightly different requirements for the spec.
```
Comment (asettle) [inline doc/mgr/orchestrator.rst]:
```suggestion
Each service type can have different requirements for the spec.

File: doc/mgr/orchestrator.rst [hunk 6]
```diff
@@ -276,6 +276,76 @@ Start/stop/reload::
...
+
+Service specifications of type ``mon``, ``mgr``, and the monitoring
+types don't require a ``service_id``
```
Comment (asettle) [inline doc/mgr/orchestrator.rst]:
```suggestion
types do not require a ``service_id``

File: doc/mgr/orchestrator.rst [hunk 7]
```diff
@@ -276,6 +276,80 @@ Start/stop/reload::
...
+    ---
+    service_type: mgr
+    placement:
+      host_pattern: mgr*
+    ---
+    service_type: osd
+    placement:
+      host_pattern: osd*
+    data_devices:
+      all: true
```
Comment (sebastian-philipp) [inline doc/mgr/orchestrator.rst:351]:
```
 yaml.scanner.ScannerError: while scanning an alias
   in "<unicode string>", line 3, column 17:
       host_pattern: *monitor*
                     ^
 expected alphabetic or numeric character, but found '*'
   in "<unicode string>", line 3, column 25:
       host_pattern: *monitor*
===
```

**Pull request merged**